arctic/arctic.py

```diff
@@ -202,6 +203,42 @@ def list_libraries(self):
                         libs.append(coll[:-1 * len(self.METADATA_COLL) - 1])
         return libs
 
+    @mongo_retry
+    def invalidate_libraries_cache(self):
+        """
+        Deletes the data in the collection used to cache the result
+        of list_libraries
+        :return:
+        """
+        self._conn.cache_db.cached_lib_names.drop()
+
+    @mongo_retry
+    def list_libraries_cached(self):
```

Comment:
I suggest caching should be a flag for the list libraries call, instead of a separate method.  Maybe sensible values are USE, REFRESH, IGNORE?

Reply:
Do you want to have some thing specific for list_libraries or a generic flag for all caching, if we do cache more stuff in the future?